test/src/test/java/hudson/model/RunMapTest.java

```diff
@@ -130,4 +140,72 @@ private Object readResolve() {
         assertEquals(0, runs.size());
     }
 
+    @Test
+    public void stream() throws Exception {
+        FreeStyleProject p = r.createFreeStyleProject();
+        SortedMap<Integer, FreeStyleBuild> builds = new TreeMap<>(Collections.reverseOrder());
+        for (int i = 0; i < 10; i++) {
+            FreeStyleBuild build = r.buildAndAssertSuccess(p);
+            builds.put(build.number, build);
+        }
+        RunMap<FreeStyleBuild> runMap = p._getRuns();
+
+        assertEquals(builds.size(), runMap.entrySet().size());
+        assertTrue(
+                runMap.entrySet().stream().spliterator().hasCharacteristics(Spliterator.DISTINCT));
+        assertTrue(
+                runMap.entrySet().stream().spliterator().hasCharacteristics(Spliterator.ORDERED));
+        assertFalse(runMap.entrySet().stream().spliterator().hasCharacteristics(Spliterator.SIZED));
+        assertFalse(
+                runMap.entrySet().stream().spliterator().hasCharacteristics(Spliterator.SORTED));
+        assertThrows(
+                IllegalStateException.class,
+                () -> runMap.entrySet().stream().spliterator().getComparator());
+
+        assertEquals(new ArrayList<>(builds.keySet()), new ArrayList<>(runMap.keySet()));
```

Comment:
```suggestion
        assertEquals(builds.keySet(), runMap.keySet());
```

fails? `Set.equals(Set)` should work no?

Reply:
That's what I thought as well, but it failed because the objects were of different types even though their contents and iteration order were the same. Some JUnit weirdness that I did not care to look into any further.

Reply:
Hmm, OK. Should not matter.